UIKitLesson1/PlayerVC.swift

```diff
@@ -0,0 +1,214 @@
+            nextTrackButton.rightAnchor.constraint(equalTo: playPauseButton.rightAnchor, constant: 85)
+        ])
+    }
+    
+    private func setupPlayer() {
+        let song = songs[position]
+        let audioPath = Bundle.main.path(forResource: song.trackName, ofType: "mp3")
+        
+        do {
+            try AVAudioSession.sharedInstance().setMode(.default)
+            try AVAudioSession.sharedInstance().setActive(true, options: .notifyOthersOnDeactivation)
+            
+            guard let audioPath = audioPath else { return }
+            player = try AVAudioPlayer(contentsOf: URL(string: audioPath)!)
+            
+            player.play()
+            
+        } catch {
+            print("error")
+        }
+        
+        setButtons()
+        
+        trackImageView.image  = UIImage(named: song.imageName)
+        albumTitleLabel.text  = song.albumName
+        trackTitleLabel.text  = song.name
+        authorTitleLabel.text = song.artistName
+        currentTimeLabel.text = "00:00"
+        durationLabel.text    = "--:--"
+        [albumTitleLabel, trackTitleLabel, authorTitleLabel].forEach{ $0.font = .boldSystemFont(ofSize: 20)}
+    }
+    
+    private func setButtons() {
+        hideButton.setImage(.init(systemName: "chevron.down"), for: .normal)
+        hideButton.addTarget(self, action: #selector(dragDownButtonTapped), for: .touchUpInside)
+        
+        shareButton.setImage(UIImage(systemName: "square.and.arrow.up"), for: .normal)
+        shareButton.addTarget(self, action: #selector(shareButtonTapped), for: .touchUpInside)
+        
+        previosTrackButton.setImage(UIImage(named: "pre"), for: .normal)
+        previosTrackButton.addTarget(self, action: #selector(previosTrack), for: .touchUpInside)
+        
+        playPauseButton.setImage(UIImage(named: "pause"), for: .normal)
+        playPauseButton.addTarget(self, action: #selector(playPouseAction), for: .touchUpInside)
+        
+        nextTrackButton.setImage(UIImage(named: "next"), for: .normal)
+        nextTrackButton.addTarget(self, action: #selector(nextTrack), for: .touchUpInside)
+        
+        currentTimeSlider.addTarget(self, action: #selector(playbackSliderValueChanged), for: .valueChanged)
+    }
+    
+    @objc func playbackSliderValueChanged(_ playbackSlider: UISlider) {
+        player.currentTime = Double(currentTimeSlider.value) * player.duration
+        updateTime()
+    }
+    
+    private func updateTime() {
+        currentTimeLabel.text = String(player.currentTime)
+        
+        let minStr: Int = Int(player.currentTime) / 60
+        let secStr: Int = Int(player.currentTime) % 60
+        if secStr > 0 && secStr <= 9  {
+            currentTimeLabel.text = String("\(minStr):0\(secStr)")
+        } else {
+            currentTimeLabel.text = String("\(minStr):\(secStr)")
+        }
+    }
+    
+    @objc func dragDownButtonTapped(_ sender: Any) {
```

Comment:
```suggestion
    @objc func dragDownButtonTapped() {
```
везде убери, если не используешь)